### PR TITLE
FeaturePolicy => PermissionPolicy

### LINKED ISSUE #22792: Feature-Policy should be called Permissions-Policy now, having a bit different syntax
**Is your feature request related to a problem? Please describe.**

[Permissions Policy, W3C Working Draft, 16 July 2020](https://www.w3.org/TR/permissions-policy-1/) specifies that `Feature-Policy` is now called `Permissions Policy`.

If Nextcloud implements these headers they should be kept up to date.

**Describe the solution you'd like**

At

https://github.com/nextcloud/server/blob/be9dde8a074117833ce7f0252ac8784146c8a9fc/lib/public/AppFramework/Http/Response.php#L243

`Permissions-Policy` should be generated instead of `Feature-Policy`.  
`Permissions-Policy` has slightly different syntax.

**Describe alternatives you've considered**

Both `Permissions-Policy` and `Feature-Policy` are generated.

**Additional context**

securityheaders.com post on the topic - [Goodbye Feature Policy and hello Permissions Policy!](https://scotthelme.co.uk/goodbye-feature-policy-and-hello-permissions-policy/)

Thanks!

### PR DESCRIPTION
**Pull request by rullzer**
We already had the FeaturePolicy header. However this call got renamed
to PermisionPolicy. Here we move this over. The old mechanism stays
there it just won't get extended. So apps that use the FeaturePolicy
will not stop to work.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

Fixes #22792

### CODE REVIEW

**Comment by MorrisJobke** [comment]
cc @LukasReschke 


🏓

File: lib/public/AppFramework/Http/EmptyFeaturePolicy.php [hunk 1]
```diff
@@ -35,6 +35,7 @@
  *
  * @see \OCP\AppFramework\Http\FeaturePolicy
  * @since 17.0.0
+ * @depreacted 21.0.0 Use \OCP\AppFramework\Http\EmptyPermissionPolicy
```
Comment (mercutiodesign) [inline lib/public/AppFramework/Http/EmptyFeaturePolicy.php:38]:
```suggestion
 * @deprecated 21.0.0 Use \OCP\AppFramework\Http\EmptyPermissionPolicy
```

(+ more instances of `depreacted` -> `deprecated`)

**Comment by MichaIng** [comment]
Wouldn't it be reasonable to map all FeaturePolicy functions to PermissionPolicy functions? So the old interface stays and throws a deprecation warning, but everything else about it can be removed and only the `Permissions-Policy` header is used. The problem is, if both headers are used, the stricter one may be applied by clients, respectively if one browser feature is denied by one of them, but allowed by the other one, it is finally denied. So if some app or NC core allows a specific feature via one of the interfaces, it is not really allowed until done with both interfaces/headers, or am I wrong?

**Comment by kesselb** [comment]
Rebased the branch.
Replaced 21 with 28. 
Fixed the namespace for AddPermissionsPolicyEvent.

**Comment by solracsf** [comment]
CI is red 🔴

File: lib/public/AppFramework/Http/FeaturePolicy.php [hunk 2]
```diff
@@ -35,6 +35,7 @@
  * should require no modification at all for most use-cases.
  *
  * @since 17.0.0
+ * @depreacted 21.0.0 use \OCP\AppFramework\Http\PermissionPolicy
```
Comment (MichaIng) [inline lib/public/AppFramework/Http/FeaturePolicy.php:38]:
```suggestion
 * @deprecated 28.0.0 use \OCP\AppFramework\Http\PermissionPolicy
```

File: lib/public/AppFramework/Http/PermissionPolicy.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * Class PermissionPolicy is a simple helper which allows applications to
+ * modify the Permission-Policy sent by Nextcloud. Per default only autoplay is allowed
+ * from the same domain and full screen as well from the same domain.
+ *
+ * Even if a value gets modified above defaults will still get appended. Please
+ * notice that Nextcloud ships already with sensible defaults and those policies
+ * should require no modification at all for most use-cases.
+ *
+ * @since 21.0.0
```
Comment (MichaIng) [inline lib/public/AppFramework/Http/PermissionPolicy.php:37]:
```suggestion
 * @since 28.0.0
```

File: lib/public/AppFramework/Http/Response.php [hunk 4]
```diff
@@ -326,13 +331,34 @@ public function getFeaturePolicy(): EmptyFeaturePolicy {
...
+		if ($this->permissionPolicy === null) {
+			$this->setPermissionPolicy(new EmptyPermissionPolicy());
+		}
+		return $this->permissionPolicy;
+	}
+
+	/**
+	 * @since 17.0.0
+	 * @depreacted 28.0.0 Use setPermissionPolicy
+	 */
```
Comment (MichaIng) [inline lib/public/AppFramework/Http/Response.php:355]:
```suggestion
	/**
	 * @since 28.0.0
	 */
```

**Comment by skjnldsv** [comment]
As there is no feedback since a while I will close this ticket. 
If you will decide to work on this feature again and if it hasn't been fixed or implemented already, feel free to re-open and solve the various conflicts.

**Pull request closed without merging**